lib/rucio/client/uploadclient.py

```diff
@@ -139,15 +139,31 @@ def upload(self, items, summary_file_path=None):
             file_did = {'scope': file['did_scope'], 'name': file['did_name']}
             dataset_did_str = file.get('dataset_did_str')
 
+            rse = file['rse']
+            rse_settings = self.rses[rse]
+            is_deterministic = rse_settings.get('deterministic', True)
+            if not is_deterministic and not pfn:
+                logger.error('PFN has to be defined for NON-DETERMINISTIC site.')
```

Comment:
Change `site` to `RSE`.

Reply:
Suggesting:
`logger.warning('Skipping upload because a PFN must be specified to upload to the non-deterministic RSE: %s' % rse)`